### PR TITLE
Refactored user history page.

### PR DESCRIPTION
**Pull request by imanherzi**
Refactored user history page so user requests and results displayed on page. 

Renamed associated files from profile to history

### CODE REVIEW

File: bait-files/src/main/java/com/google/sps/servlets/UserHistoryServlet.java [hunk 1]
```diff
@@ -47,26 +53,26 @@ public int compare(Review a, Review b) {
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     boolean isPendingReview = Boolean.parseBoolean(request.getParameter("is-pending-review"));
     if (isPendingReview) {
-      List<Review> pendingReviews = getReviewsByType(Status.PENDING);
+      List<Request> allRequests = ObjectifyService.ofy().load().type(Request.class).list();
+      List<Request> requests = allRequests.stream()
+                                   .filter(req -> req.userId.equals(userId))
+                                   .filter(req -> req.status == Status.PENDING)
+                                   .collect(toList());
+      Collections.sort(requests, new SortByEarliestDate());
```
Comment (andrewseamanco) [inline bait-files/src/main/java/com/google/sps/servlets/UserHistoryServlet.java:61]: You can replace this with this sort
`    List<Review> reviews = allRequests.stream()
                               .filter(req -> req.userId.equals(userId))
                               .filter(req -> req.status == statusType)
                               .sorted(Comparator.comparing(req -> req.submissionDate)
                               .collect(toList());`

Comment (andrewseamanco) [reply bait-files/src/main/java/com/google/sps/servlets/UserHistoryServlet.java:61]: If in the wrong direction you can add .reversed() to the comparator

File: bait-files/src/main/java/com/google/sps/servlets/UserHistoryServlet.java [hunk 2]
```diff
@@ -47,26 +53,26 @@ public int compare(Review a, Review b) {
...
+      response.getWriter().println(new Gson().toJson(requests));
       return;
     } else {
-      List<Review> completedReviews = getReviewsByType(Status.COMPLETED);
+      List<Review> allReviews = ObjectifyService.ofy().load().type(Review.class).list();
+      List<Review> reviews = allReviews.stream()
+                                 .filter(review -> review.userId.equals(userId))
+                                 .filter(review -> review.status == Status.COMPLETED)
+                                 .collect(toList());
+      Collections.sort(reviews, new SortByMostRecentDate());
```
Comment (andrewseamanco) [inline bait-files/src/main/java/com/google/sps/servlets/UserHistoryServlet.java:71]: Same replacement as above

File: bait-files/src/main/java/com/google/sps/servlets/UserHistoryServlet.java [hunk 3]
```diff
@@ -26,47 +26,25 @@
...
   UserService userService = UserServiceFactory.getUserService();
   String userId = userService.getCurrentUser().getUserId();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     boolean isPendingReview = Boolean.parseBoolean(request.getParameter("is-pending-review"));
     if (isPendingReview) {
-      List<Review> pendingReviews = getReviewsByType(Status.PENDING);
+      List<Request> allRequests = ObjectifyService.ofy().load().type(Request.class).list();
+      List<Request> requests = allRequests.stream() .filter(req -> req.userId.equals(userId)) .filter(req -> req.status == Status.PENDING).sorted(Comparator.comparing(req -> req.submissionDate)) .collect(toList());;
```
Comment (kzweller) [inline bait-files/src/main/java/com/google/sps/servlets/UserHistoryServlet.java:37]: This needs some formatting.

**Pull request merged**